README.md

```diff
@@ -108,19 +108,25 @@ This weirdness is tracked in [issue 100](https://github.com/include-what-you-use
 
 The original design was built for Make, but a number of alternative run modes have come up over the years.
 
-#### Plugging into Make ####
+#### Running on single source file ####
 
-The easiest way to run IWYU over your codebase is to run
+The simplest way to use IWYU is to run it against a single source file:
 
-      make -k CXX=/path/to/llvm/Debug+Asserts/bin/include-what-you-use
+      include-what-you-use $CXXFLAGS myfile.cc
 
-or
+where `$CFLAGS` are the flags you would normally pass to the compiler.
 
-      make -k CXX=/path/to/llvm/Release/bin/include-what-you-use
+#### Plugging into existing build system ####
 
-(include-what-you-use always exits with an error code, so the build system knows it didn't build a .o file.  Hence the need for `-k`.)
+The simplest example above doesn't scale very well to projects with many source files, so typically you would replace the compiler in your build system with `include-what-you-use` to generate a large batch of IWYU advice.
 
-Include-what-you-use only analyzes .cc (or .cpp) files built by `make`, along with their corresponding .h files.  If your project has a .h file with no corresponding .cc file, IWYU will ignore it unless you use the `--check_also` switch to add it for analysis together with a .cc file.
+Depending on your build system/build tools, this can take many forms, but for a simple GNU Make system it might look like this:
+
+      make -k CXX=include-what-you-use EXTRA_CXXFLAGS="-Xiwyu --fail"
+
+(The additional `-Xiwyu --fail` switch makes `include-what-you-use` always exit with an error code, so the build system knows it didn't build a .o file.  Hence the need for `-k`.)
+
+Include-what-you-use only analyzes .cc (or .cpp) files built by your build system, along with their corresponding .h files.  If your project has a .h file with no corresponding .cc file, IWYU will ignore it unless you use the `--check_also` switch to add it for analysis together with a .cc file.
```

Comment:
Not really.  I've been analyzing header files as well as source files in the "simple" way:

```
alx@ady1:~/src/test$ cat foo.h 
#include <stdio.h>

int foo(void);
```
```
alx@ady1:~/src/test$ iwyu foo.h 

foo.h should add these lines:

foo.h should remove these lines:
- #include <stdio.h>  // lines 1-1

The full include-list for foo.h:
---
```

Reply:
Nice, yes. We did that kind of headers-first analysis at a previous workplace too, and it really helped reduce header dependencies (which are the worst contributors to build time).

I think what the IWYU docs are getting at is that `include-what-you-use` is primarily a compiler replacement running in the context of a build system. It doesn't have a way to figure out flags (`CFLAGS` or `CXXFLAGS`) for any arbitrary header. And many larger C++ projects depend on their build systems in order for any source code to compile. I'll think about better phrasing here, I agree that the current isn't ideal.

Reply:
Added a little sentence on analyzing headers.